### PR TITLE
Move to eslint v9

### LINKED ISSUE #1: ESLint v9.0.0 Support
Eslint v9.0.0 has been released for a while and has gained quite some support from ESLint plugins. I suggest to upgrade to ESLint v9.0.0, as this packs a lot of features that are very useful for such a repository.

In general; I would suggest an approach where you provide multiple configs for different types of projects; prime examples being `vue` and `react`, but I would also split out plain `javascript` and `typescript` as separate configs. In another repository, one can then simply import all the config files they require and add them to their personal ESLint config.

As an example, take a look at [this ESLint config](https://github.com/GEWIS/aurora-backoffice/blob/Feature/ui-overhaul/eslint.config.mjs). This kind of follows this principle, but then takes configs from other packages.

### PR DESCRIPTION
**Pull request by JustSamuel**
Based on Gijs' config

Closes #1

### CODE REVIEW

**Comment by Gijsdeman** [comment]
Merging is fine on my behalf, but this config will not give desired results in every repository, nor does it close #1. 

- The config still needs to be extended with plenty of options that have been used over other repositories
- The config should, in my opinion, still be split in separate pieces that can be chosen from by the end user. Most obvious `vue` and `react`, since you will likely not use both in the same project. But also think of specific items like `i18n`, that are simply not part in all projects. I would even suggest making separate configs for `frontend` and `backend`, since general styling/linting between those two might differ as well.

I can take a look somewhere in the future to make this a reality, as I think on all fronts it is more favorable to provide multiple configs that people can combine in a standardized way than provide a single config and have people deal with the issues that might come along with it.

File: eslint.common.mjs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import eslint from '@eslint/js';
+import tsEslint from 'typescript-eslint';
+import importPlugin from 'eslint-plugin-import';
+
+import {includeIgnoreFile} from "@eslint/compat";
+import path from "node:path";
+import {fileURLToPath} from "node:url";
```
Comment (JustSamuel) [inline eslint.common.mjs:7]: spacing around {

**Review by JustSamuel — COMMENTED** [review]
Maybe also and an example to the readme how to use the other configs?

File: eslint.common.mjs (hunk 1 above)
Comment (Gijsdeman) [reply eslint.common.mjs]: This part of code is mostly removed to actually fetch `.gitignore` from root directories.

**Comment by Gijsdeman** [comment]
On my part this is ready to merge. Config files seem to be working nicely for both `react` and `vue`. Minor changes can be made later down the line.

**Review by Gijsdeman — APPROVED** [review]

**Pull request merged**